### Tareas

- Diagrama entidad relacion (Daro)
- Analisis exploratorio de los datos (Pablo y Ana)
- Propuesta de indicadores + prototipo de dashboard (Noe y Esteban)

# 0. Librerias

In [ ]:
# Manejo de archivos y operaciones matemáticas
import math
import os

# Manipulación y análisis de datos
import numpy as np
import pandas as pd

# Visualización de datos
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

#Constantes
from  constants import DATA_RAW


# 1. Ingesta de datos 

In [2]:
df = pd.read_csv(DATA_RAW)

# 2. Exploración inicial (Initial EDA)

## Reviso Estrucutra

In [3]:
df.head(5)

,Employee_ID,Full_Name,Department,Job_Title,Hire_Date,Performance_Rating,Experience_Years,Status,Work_Mode,Salary,Year,Country,City,Age,Job_Level
0,EMP0000001,Heinz-Georg Eimer,Sales,Business Development,2023-01-31,Satisfactory,8,Active,On-site,92992.0,2023,Germany,Munich,32,Mid
1,EMP0000002,Maartje van den Nuwenhuysen-Geertsen,HR,HR Manager,2008-11-07,Good,11,Active,On-site,112318.0,2008,Netherlands,Eindhoven,43,Senior
2,EMP0000003,Sara Sureda Figueroa,HR,Talent Specialist,2016-03-19,Needs Improvement,15,Active,On-site,111121.0,2016,Spain,Seville,43,Senior
3,EMP0000004,Luce Sanchez,Operations,Operations Director,2024-04-03,Good,1,Active,Hybrid,49012.0,2024,France,Marseille,23,Junior
4,EMP0000005,William Jennings,Sales,Regional Lead,2024-11-17,Good,2,Active,On-site,57553.0,2024,United Kingdom,Edinburgh,31,Junior


In [4]:
df.tail(5)

,Employee_ID,Full_Name,Department,Job_Title,Hire_Date,Performance_Rating,Experience_Years,Status,Work_Mode,Salary,Year,Country,City,Age,Job_Level
1999995,EMP1999996,Ottmar Kusch-Dörschner,IT,Software Developer,2020-11-30,Good,15,Active,Hybrid,136684.0,2020,Germany,Frankfurt,44,Senior
1999996,EMP1999997,Isis Wagenvoort,Sales,Sales Manager,2024-06-06,Good,3,Active,On-site,57911.0,2024,Netherlands,Amsterdam,25,Junior
1999997,EMP1999998,Ángel Gomez Arenas,Sales,Sales Manager,2020-05-21,Excellent,14,Active,Remote,146065.0,2020,Spain,Valencia,39,Senior
1999998,EMP1999999,Carmela Weiß,Sales,Business Development,2018-07-11,Excellent,9,Active,Remote,138855.0,2018,Germany,Frankfurt,35,Senior
1999999,EMP2000000,Macario Casals,IT,Software Developer,2025-09-11,Good,1,Active,Hybrid,46067.0,2025,Spain,Madrid,28,Junior


In [5]:
df.columns

Index(['Employee_ID', 'Full_Name', 'Department', 'Job_Title', 'Hire_Date',
       'Performance_Rating', 'Experience_Years', 'Status', 'Work_Mode',
       'Salary', 'Year', 'Country', 'City', 'Age', 'Job_Level'],
      dtype='str')

In [6]:
df.shape

(2000000, 15)

In [7]:
df.dtypes

Employee_ID               str
Full_Name                 str
Department                str
Job_Title                 str
Hire_Date                 str
Performance_Rating        str
Experience_Years        int64
Status                    str
Work_Mode                 str
Salary                float64
Year                    int64
Country                   str
City                      str
Age                     int64
Job_Level                 str
dtype: object

In [8]:
# Revisar con el equipo  -- Dejo esto para futuro diccionario  de arreglo de tipos
"""
Employee_ID               str
Full_Name                 str
Department                str
Job_Title                 str
Hire_Date                 str YYYY/MM/DD --| Datetime
Performance_Rating        str
Experience_Years        int64
Status                    str
Work_Mode                 str
Salary                float64 
Year                    int64  --| Lo dejamos asi?
Country                   str
City                      str
Age                     int64
Job_Level                 str

"""

'\nEmployee_ID               str\nFull_Name                 str\nDepartment                str\nJob_Title                 str\nHire_Date                 str YYYY/MM/DD --| Datetime\nPerformance_Rating        str\nExperience_Years        int64\nStatus                    str\nWork_Mode                 str\nSalary                float64 \nYear                    int64  --| Lo dejamos asi?\nCountry                   str\nCity                      str\nAge                     int64\nJob_Level                 str\n\n'

## Revision de Nulos 

In [9]:
df.isnull().sum().sort_values(ascending=False)


Performance_Rating    3333
Full_Name                0
Employee_ID              0
Department               0
Job_Title                0
Hire_Date                0
Experience_Years         0
Status                   0
Work_Mode                0
Salary                   0
Year                     0
Country                  0
City                     0
Age                      0
Job_Level                0
dtype: int64

Reviso si existe algun otro tipo de nulo que este formateado como un string nan,0,-

In [10]:
columns=['Department', 'Job_Title', 
       'Performance_Rating', 'Experience_Years', 'Status', 'Work_Mode',
       'Salary', 'Year', 'Country', 'City', 'Age', 'Job_Level']

for col in columns:
    print("\n", col)
    print(df[col].value_counts(dropna=False))
    print("-------------------------------------------")


 Department
Department
Sales         600633
IT            499325
Operations    400595
Finance       299955
HR            199492
Name: count, dtype: int64
-------------------------------------------

 Job_Title
Job_Title
Sales Manager           120595
Sales Executive         120349
Sales Director          119971
Regional Lead           119886
Business Development    119832
IT Manager              100132
System Admin            100103
DevOps Engineer          99886
Data Engineer            99747
Software Developer       99457
Operations Director      80346
Logistics                80242
Operations Manager       80119
Supply Chain             80025
Operations Analyst       79863
Accountant               60125
Tax Specialist           60050
Financial Analyst        60017
CFO                      59921
Finance Manager          59842
HR Coordinator           40070
HR Manager               39989
HR Director              39887
Talent Specialist        39848
Recruiter                39698
Name

REVISAR CON EL EQUIPO: 
Resumen de análisis de nulos 

1) Performance_Rating  --- 3333 Null values
2) años de experiencia negativos
3) Salarios negativos y atipicos revisar
4) Experience years negativos

## Revision de Duplicados

In [11]:
# Reviso si hay filas duplicadas 
df[df.duplicated(keep=False)]

,Employee_ID,Full_Name,Department,Job_Title,Hire_Date,Performance_Rating,Experience_Years,Status,Work_Mode,Salary,Year,Country,City,Age,Job_Level


In [12]:
# Reviso si hay Ids duplicados
df[df.duplicated(subset=["Employee_ID"], keep=False)].sort_values(by="Employee_ID")


,Employee_ID,Full_Name,Department,Job_Title,Hire_Date,Performance_Rating,Experience_Years,Status,Work_Mode,Salary,Year,Country,City,Age,Job_Level


In [13]:
# ver empleados duplicados
df[df.duplicated(subset=["Full_Name"], keep=False)].sort_values("Full_Name")

,Employee_ID,Full_Name,Department,Job_Title,Hire_Date,Performance_Rating,Experience_Years,Status,Work_Mode,Salary,Year,Country,City,Age,Job_Level
1613661,EMP1613662,Aaliyah Ansems,IT,Software Developer,2020-01-08,Good,1,Active,On-site,59408.0,2020,Netherlands,The Hague,30,Junior
1233693,EMP1233694,Aaliyah Ansems,IT,Data Engineer,2022-11-26,Good,2,Active,Remote,53005.0,2022,Netherlands,Amsterdam,25,Junior
1651398,EMP1651399,Aaliyah Arens,Sales,Sales Executive,2023-05-31,Excellent,6,Active,On-site,96787.0,2023,Netherlands,Amsterdam,28,Mid
1858581,EMP1858582,Aaliyah Arens,Operations,Operations Analyst,2025-09-30,Excellent,0,Active,Hybrid,50928.0,2025,Netherlands,Eindhoven,21,Junior
979926,EMP0979927,Aaliyah Bresse,Operations,Logistics,2023-10-21,Good,4,Active,On-site,90978.0,2023,Netherlands,Utrecht,31,Mid
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1086330,EMP1086331,Łukasz Sugier,IT,DevOps Engineer,2025-08-14,Excellent,3,Active,Hybrid,56234.0,2025,Poland,Warsaw,23,Junior
1272116,EMP1272117,Łukasz Swiątek,IT,IT Manager,2018-08-09,Excellent,11,Active,On-site,138440.0,2018,Poland,Warsaw,31,Senior
1974494,EMP1974495,Łukasz Swiątek,Operations,Operations Manager,2025-06-06,Satisfactory,2,Active,Remote,40386.0,2025,Poland,Warsaw,25,Junior
1229255,EMP1229256,Łukasz Wójcikiewicz,Operations,Operations Analyst,2025-01-18,Satisfactory,6,Active,On-site,93919.0,2025,Poland,Krakow,33,Mid


REVISAR: Hablar con el equipo de las reglas de negocio para poder avanzar con el tema de logicas. Revisar con Ana los sueldos para ver si existe alguna inconsistencia 

In [14]:
df.columns

Index(['Employee_ID', 'Full_Name', 'Department', 'Job_Title', 'Hire_Date',
       'Performance_Rating', 'Experience_Years', 'Status', 'Work_Mode',
       'Salary', 'Year', 'Country', 'City', 'Age', 'Job_Level'],
      dtype='str')

## 5. Outliers

Analísis de cuartiles

6. Consistencia



6. Consistencia
7. Rangos
8. Relaciones
9. Claves
10. Distribución
11. Irrelevantes
12. Calidad / sesgos

# 4. Limpieza de datos (Data Cleaning)

# 5. Análisis exploratorio (EDA) - Analítica


In [ ]:
df_grouped = df.groupby(['Department', 'Job_Title']).size().reset_index(name='Total_Employees')
df_grouped 


Posible Organigrama
![Organigrama](Bosquejo_Organigrama.png)

# 6. Visualización de datos (Data Visualization)

# 7. Generación de insights (Conclusiones)